Runtime/Highlight/DefaultHighlighter.cs

```diff
@@ -0,0 +1,445 @@
+﻿using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Rendering;
+using Innoactive.Creator.Unity;
+
+namespace Innoactive.Creator.BasicInteraction
+{
+    /// <summary>
+    /// Adds highlighting functionality to a GameObject with Renderers.
+    /// </summary>
+    /// <remarks>
+    /// Highlights are always queued following a LIFO (Last In First Out) scheme. 
+    /// </remarks>
+    [DisallowMultipleComponent]
+    public class DefaultHighlighter : MonoBehaviour
+    {
+        private readonly Dictionary<string, Material> activeHighlights = new Dictionary<string, Material>();
```

Comment:
I dont understand why this is a dictionary, when it actually should be a stack/list

Reply:
On it, the change requires some refactoring and testing.